shotover-proxy/tests/kafka_int_tests/test_cases.rs

```diff
@@ -158,10 +169,123 @@ pub async fn produce_consume_partitions1(
+                Some(0),
+            )
+            .await;
+
+        let mut consumer = connection_builder
+            .connect_consumer(topic_name, "consumer_group_using_offsets")
+            .await;
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "Initial".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(0),
+            })
+            .await;
+
+        // The offset to be committed should be lastProcessedMessageOffset + 1
+        let offset1 = HashMap::from([(
+            TopicPartition {
+                topic_name: topic_name.to_owned(),
+                partition: 0,
+            },
+            1,
+        )]);
+        consumer.assert_commit_offsets(offset1);
+
+        producer
+            .assert_produce(
+                Record {
+                    payload: "Message1",
+                    topic_name,
+                    key: Some("Key"),
+                },
+                Some(1),
+            )
+            .await;
+
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "Message1".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(1),
+            })
+            .await;
+
+        let offset2 = HashMap::from([(
+            TopicPartition {
+                topic_name: topic_name.to_owned(),
+                partition: 0,
+            },
+            2,
+        )]);
+        consumer.assert_commit_offsets(offset2);
+
+        producer
+            .assert_produce(
+                Record {
+                    payload: "Message2",
+                    topic_name,
+                    key: Some("Key"),
+                },
+                Some(2),
+            )
+            .await;
+    }
+
+    {
+        // The new consumer should consume Message2 which is at the last uncommitted offset
+        let mut consumer = connection_builder
+            .connect_consumer(topic_name, "consumer_group_using_offsets")
+            .await;
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "Message2".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(2),
+            })
+            .await;
+    }
+
+    {
+        // The new consumer should still consume Message2 as its offset has not been committed
+        let mut consumer = connection_builder
+            .connect_consumer(topic_name, "consumer_group_using_offsets")
+            .await;
+        consumer
+            .assert_consume(ExpectedResponse {
+                message: "Message2".to_owned(),
+                key: Some("Key".to_owned()),
+                topic_name: topic_name.to_owned(),
+                offset: Some(2),
+            })
+            .await;
+    }
+}
+
```

Comment:
Add another test case for consumer in "consumer_group_without_offsets" that should consume the Initial message.